common/djangoapps/student/views.py

```diff
@@ -1670,16 +1670,14 @@ def create_account_with_params(request, params):
         getattr(settings, 'REGISTRATION_EXTRA_FIELDS', {})
     )
 
-    # Boolean of whether a 3rd party auth provider and credentials were provided in
-    # the API so the newly created account can link with the 3rd party account.
-    #
-    # Note: this is orthogonal to the 3rd party authentication pipeline that occurs
-    # when the account is created via the browser and redirect URLs.
-    should_link_with_social_auth = third_party_auth.is_enabled() and 'provider' in params
-
-    if should_link_with_social_auth or (third_party_auth.is_enabled() and pipeline.running(request)):
```

Comment:
Since the previous comments are now lost, for future readability and understanding of this code, can you add a self-explanatory temporary variable, like follows?
`third_party_auth_credentials_in_api = 'provider' in params`
And then, replace the 2 places below that say `'provider' in params` to `third_party_auth_credentials_in_api`.